### ISSUE TITLE
[BUG] Including hidden files also includes (git)ignored files if the .gitignore is in the parent directory

### ISSUE THREAD

**Issue by taeruh** [sharkdp/fd#1461]
### Checks

- [X] I have read the troubleshooting section and still think this is a bug.

### Describe the bug you encountered:

Assume we have the following directory structure:
```
.
├── .abc
├── .gitignore
└── sub_directory
   └── .abc
```
where
```
cat .gitignore
.abc
```
Using `fd` in the `sup_directory` with `--hidden`, I get
```
fd --hidden
.abc
```
The `.abc` file is shown, although it should have been ignored.

### Describe what you expected to happen:

Do not show the `.abc` file.

Note that this only happens in the `sup_directory`. Using `fd --hidden` in the root directory shows the expected output.

Possible Fix:
```
diff --git a/src/cli.rs b/src/cli.rs
index 0007a31..3a7081f 100644
--- a/src/cli.rs
+++ b/src/cli.rs
@@ -643,7 +643,7 @@ impl Opts {
         } else if !self.search_path.is_empty() {
             &self.search_path
         } else {
-            let current_directory = Path::new(".");
+            let current_directory = Path::new("./");
             ensure_current_directory_exists(current_directory)?;
             return Ok(vec![self.normalize_path(current_directory)]);
         };
```
All tests pass in the github workflows with this change.

The bug is not really caused by something `fd` is doing but rather by what the "Walker" in the `ignore` dependency is doing. A simple loop over `WalkBuilder::new(path).hidden(false).build()` with `path = "."` and `path = "./"` leads to the same wrong and correct results, respectively. Maybe one could report it as a bug there? I'm rather confused by this behavior though.

Anyways, the diff above should fix it in `fd`, I think (could open a PR if you want).

### What version of `fd` are you using?

fd 9.0.0

### Which operating system / distribution are you on?

```shell
Linux 6.1.69-1-lts x86_64
```

**Comment by tavianator** [comment]
This definitely seems like a bug in the `ignore` crate.  You should have a `.git` folder in your reproducer, but even with that it fails:

```console
tavianator@graphene $ tree -a .
.
├── .abc
├── .git
├── .gitignore
└── sub_directory
    └── .abc

2 directories, 4 files
tavianator@graphene $ cat .gitignore
.abc
tavianator@graphene $ cd sub_directory
tavianator@graphene $ fd --hidden
.abc
tavianator@graphene $ fd --hidden --search-path ./
tavianator@graphene $ rg --hidden --files .
./.abc
tavianator@graphene $ rg --hidden --files ./
tavianator@graphene $ 
```

Since `rg` by itself can reproduce the bug, it should be reported to https://github.com/BurntSushi/ripgrep (unless there's already a duplicate, but I didn't find one quickly).

**Comment by tmccombs** [comment]
This seems somewhat similar to https://github.com/BurntSushi/ripgrep/issues/829.

**Comment by taeruh** [comment]
Sorry for not responding earlier; I wanted to dig a little bit into it and hadn't had the time to do that before.

The problems might be similar, but I don't think they are directly related. I think I tracked down the problem for the hidden files in the ignore crate, and made a PR there ([BurntlSushi/ripgrep#2711](https://github.com/BurntSushi/ripgrep/pull/2711)).

In the meantime, until it gets fixed in ripgrep, one could apply the fix I proposed above?
